File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
import sys
input = sys.stdin.readline
N = int(input())
new_user = False
count = 0
chat_user = set()
for i in range(N) :
    chat = input().rstrip()

    if chat == "ENTER" :
        new_user = True
        if len(chat_user) > 0 :
            chat_user.clear()
    else :
        if new_user and chat not in chat_user :
            chat_user.add(chat)
            count += 1

print(count)

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const [n, ...input] = fs.readFileSync(filePath).toString().split("\n");

let answer = 0;
let people_set = new Set();

for(let i = 0;i < n;i++){
    if(input[i] == 'ENTER') {
        answer += people_set.size;
        people_set.clear();
    } else if(!people_set.has(input[i])){
        people_set.add(input[i]);
    }
}
console.log(answer + people_set.size)